modules/xfeatures2d/src/daisy.cpp

```diff
@@ -0,0 +1,2057 @@
+            return true;
+         return false;
+      default:
+         if( is_outside(x,roi.lx,roi.ux,le,ue) || is_outside(y,roi.ly,roi.uy,le,ue) )
+            return true;
+         return false;
+      }
+    }
+
+    // computes the square of a number and returns it.
+    template<class T> inline
+    T square(T a)
+    {
+      return a*a;
+    }
+
+    // computes the square of an array. if in_place is enabled, the
+    // result is returned in the array arr.
+    template<class T> inline
+    T* square(T* arr, int sz, bool in_place=false)
+    {
+      T* out;
+      if( in_place ) out = arr;
+      else           out = allocate<T>(sz);
+
+      for( int i=0; i<sz; i++ )
+         out[i] = arr[i]*arr[i];
+
+      return out;
+    }
+
+    // computes the l2norm of an array: [ sum_i( [a(i)]^2 ) ]^0.5
+    template<class T> inline
+    float l2norm( T* a, int sz)
+    {
+      float norm=0;
+      for( int k=0; k<sz; k++ )
+         norm += a[k]*a[k];
+      return sqrt(norm);
+    }
+
+    // computes the l2norm of the difference of two arrays: [ sum_i( [a(i)-b(i)]^2 ) ]^0.5
+    template<class T1, class T2> inline
+    float l2norm( T1* a, T2* b, int sz)
+    {
+      float norm=0;
+      for( int i=0; i<sz; i++ )
+      {
+         norm += square( (float)a[i] - (float)b[i] );
+      }
+      norm = sqrt( norm );
+
+      return norm;
+    }
+
+    template<class T> inline
+    float l2norm( T y0, T x0, T y1, T x1 )
+    {
+      float d0 = (float) (x0 - x1);
+      float d1 = (float) (y0 - y1);
+
+      return sqrt( d0*d0 + d1*d1 );
+    }
+
+    // allocates a memory of size sz and returns a pointer to the array
+    template<class T> inline
+    T* allocate(const int sz)
+    {
+      T* array = new T[sz];
+      return array;
+    }
+
+    // allocates a memory of size ysz x xsz and returns a double pointer to it
+    template<class T> inline
+    T** allocate(const int ysz, const int xsz)
+    {
+      T** mat = new T*[ysz];
+      int i;
+
+      for(i=0; i<ysz; i++ )
+         mat[i] = new T[xsz];
+      // allocate<T>(xsz);
+
+      return mat;
+    }
+
+    // deallocates the memory and sets the pointer to null.
+    template<class T> inline
+    void deallocate(T* &array)
+    {
+      delete[] array;
+      array = NULL;
+    }
+
+    // deallocates the memory and sets the pointer to null.
+    template<class T> inline
+    void deallocate(T** &mat, int ysz)
+    {
+      if( mat == NULL ) return;
+
+      for(int i=0; i<ysz; i++)
+         deallocate(mat[i]);
+
+      delete[] mat;
+      mat = NULL;
+    }
+
+    // Converts the given polar coordinates of a point to cartesian
+    // ones.
+    template<class T1, class T2> inline
+    void polar2cartesian(T1 r, T1 t, T2 &y, T2 &x)
+    {
+      x = (T2)( r * cos( t ) );
+      y = (T2)( r * sin( t ) );
+    }
+
+
+    template<typename T> inline
+    void convolve_sym_( T* image, int h, int w, T* kernel, int ksize )
```

Comment:
[DONE], commented already (upper) on the conv issue, that template too is now gone.